File: 997-Find-The-Town-Judge/findJudge.java
class Solution {
  public int findJudge(int N, int[][] trust) {
      int[] scores = new int[N+1];
      
      for (int[] relation : trust) {
          scores[relation[0]]--;
          scores[relation[1]]++;
      }
      
      for (int i = 1; i < N + 1; i++) {
          if (scores[i] == N - 1) return i;
      }
      
      return -1;
  }
}

File: 997-Find-The-Town-Judge/findJudge.py
class Solution:
    def findJudge(self, N: int, trust: List[List[int]]) -> int:
        scores = [0] * (N + 1)
        
        for truster, trustee in trust:
            scores[truster] -= 1
            scores[trustee] += 1
            
        for i in range(1, N + 1):
            if scores[i] == N - 1: return i
        
        return -1